File: Mechanics/Unit/TownHall.py
from Mechanics.Constants import Config
from Mechanics.Unit import Unit


class TownHall(Unit.Unit):

    # Identifier
    id = Unit.TOWN_HALL
    structure = Unit.STRUCTURE

    # Tile Data
    width = 3
    height = 3

    # Economics
    cost_lumber = 800
    cost_gold = 1200
    build_duration = 255 * Config.FRAME_MULTIPLIER

    # Stats
    name = "Town Hall"
    damage_min = 0
    damage_max = 0
    damage_piercing = 0
    damage_range = 0
    armor = 0
    health = 1200
    health_max = 1200
    speed = 0
    sight = 4
    food = 5
    food_cost = 0


    def __init__(self, player, attrs={}):
        super().__init__(player, attrs)
        self.player = player

        self.buildable = [
            self.game.UnitManager.UNITS[self.game.UnitManager.WORKER][player.race]
        ]

File: src/game/unit/TownHall.cpp
//
// Created by Per-Arne on 24.02.2017.
//

#include "TownHall.h"
#include "../player/Player.h"

TownHall::TownHall(Player &player): Unit(player) {
    typeId = Constants::Unit::TownHall;
    health = 1200;
    health_max = 1200;
    direction = 0;


    groundUnit = true;
    waterUnit = false;

    damageMin = 0;
    damageMax = 0;
    damageRange = 0;
    damagePiercing = 0;
    armor = 0;

    goldCarry = 0;
    lumberCarry = 0;
    oilCarry = 0;
    carryCapacity = 0;
    speed = 0;
    sight = 4;

    canHarvest = false;
    canAttack = false;
    canMove = false;
    military = false;
    structure = true;
    recallable = true;

    lumberCost = 250;
    goldCost = 500;
    oilCost = 0;

    spawnDuration = 1; //255 * Config::getInstance().getTickModifier();

    foodProduction = 1;
    foodConsumption = 0;

    width = 3;
    height = 3;

    name = "Town-Hall";

    buildInventory = player_.inventoryManager.townhall;

}
